File: mvp_app/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/section/SectionListener.java
package com.tosmart.xiexuming.libtsparser.section;

/**
 * @author xiexuming
 */
public interface SectionListener {
    /**
     * 组建pat
     *
     * @param programAssociationSection pas
     */
    void makePat(ProgramAssociationSection programAssociationSection);

    /**
     * 组建pat
     * @param pid pid
     * @param programMapSection 特定section
     */
    void makePmt(int pid, ProgramMapSection programMapSection);

    /**
     * 组建sdt
     * @param serviceDescriptionSection 特定section
     */
    void makeSdt(ServiceDescriptionSection serviceDescriptionSection);
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/section/SectionListener.kt
package com.example.libtosmartparse.section

/**
 * @author xxm
 * 2022/7/17
 **/
open interface SectionListener {
    /**
     * 组建pat
     *
     * @param programAssociationSection pas
     */
    fun makePat(programAssociationSection: ProgramAssociationSection?)

    /**
     * 组建pat
     * @param pid pid
     * @param programMapSection 特定section
     */
    fun makePmt(pid: Int, programMapSection: ProgramMapSection?)

    /**
     * 组建sdt
     * @param serviceDescriptionSection 特定section
     */
    fun makeSdt(serviceDescriptionSection: ServiceDescriptionSection?)
}